feat: allow template strings

scottcanoni opened the pull request:
This change allows the use of template strings with variables within translated configurations.  It looks for translations that explicitly begin with:
```
"`
```
and end with
```
`"
```

Configuration JSON:
```
{
  "HelloMessage": "`Hello ${userName}`"
}
```
Usage:
```
const userName = 'International User';
const message = __('HelloMessage');
```

Expected output when rendering message:
`Hello International User`

<!--
1. [Read and sign the CLA](https://cla.js.foundation/webpack/webpack.js.org). This needs to be done only once. PRs that haven't signed it won't be accepted.
2. Check out the [development guide](https://webpack.js.org/development/) for the API and development guidelines.
3. Read through the PR diff carefully as sometimes this can reveal issues. The work will be reviewed, but this can save some effort.
-->

scottcanoni committed:
Allowing template strings

This change allows the use of template strings with variables within translated configurations.  It looks for translations that explicitly begin with `"\`` and end with `\`"`

Configuration JSON:
```
{
  "HelloMessage": "`Hello ${userName}`"
}
```
Usage:
```
const userName = 'International User';
const message = __('HelloMessage');
```

Expected output when rendering message:
`Hello International User`

alexander-akait reviewed (changes requested):
Can we add tests?

scottcanoni committed:
Adding unit tests for template string translations

scottcanoni committed:
Adding unit tests for template string translations

scottcanoni commented:
I think I've added the appropriate tests for template strings.  Please let me know what you think.  Thank you kindly.

michael-ciniawsky commented:
@scottcanoni Does the email address you're currently using for your github account and the one you use in your `.gitconfig` differ ? The CLA Bot thinks you're two different entities in this case and you need to change the commit author of your commits

michael-ciniawsky commented on test/cases/apply-translations.test.js:
```diff
@@ -6,6 +6,7 @@ describe('apply-translations', () => {
   beforeAll(() => {
     const translations = {
       'static-key': 'translated static key',
+      'template-string': '`Your dynamic value is ${variableForTemplateString}.`', // eslint-disable-line no-template-curly-in-string
```
Please move the `eslint-disable` comment to the top 

```js
/* eslint-disable
  no-template-curly-in-string,
*/
```

michael-ciniawsky reviewed (approved):
👍  Thx

scottcanoni committed:
Moving eslint disable to the top

scottcanoni commented:
@michael-ciniawsky I think the CLA bot just took time to re-check.  I added my additional email address and verified it under my Github profile and after some time, I see that the CLA bot has marked it as signed.

Also, I moved the eslint disable comment to the top of the file.

scottcanoni committed:
Fixing lint error

scottcanoni commented:
@evilebottnawi are the tests that I added adequate?

I am not sure how to resolve this issue on the continuous integration environment:
`The command "npm --version" failed and exited with 1 during .`

michael-ciniawsky commented:
> ERROR: npm is known not to run on Node.js v4.3.2

That's unrelated to this PR and should be fixed by #64 instead. Just needs sign-off from 1-2 more maintainers and g2g imho

scottcanoni commented:
@evilebottnawi and @michael-ciniawsky, now that the unit tests are there, can this be merged?
